app/javascript/packages/memorable-date/index.ts

```diff
@@ -0,0 +1,386 @@
+/**
+ * Keys to lookup error messages for different states
+ * of the MemorableDate element
+ */
+export enum MemorableDateErrorMessage {
+  missing_month_day_year = 'missing_month_day_year',
+  missing_month_day = 'missing_month_day',
+  missing_month_year = 'missing_month_year',
+  missing_day_year = 'missing_day_year',
+  missing_month = 'missing_month',
+  missing_day = 'missing_day',
+  missing_year = 'missing_year',
+  invalid_month = 'invalid_month',
+  invalid_day = 'invalid_day',
+  invalid_year = 'invalid_year',
+  invalid_date = 'invalid_date',
+  range_underflow = 'range_underflow',
+  range_overflow = 'range_overflow',
+  outside_date_range = 'outside_date_range',
+}
+
+/**
+ * Type for a range check with a corresponding error message
+ */
+interface RangeErrorMessage {
+  min?: string;
+  max?: string;
+  message: string;
+}
+
+/**
+ * Type for a hash in which the specified messages can be looked up
+ */
+type MemorableDateErrorMessageLookup = Record<
+  MemorableDateErrorMessage & string,
+  string | undefined
+>;
+
+interface ErrorMessageLookupContainer {
+  error_messages: MemorableDateErrorMessageLookup;
```

Comment:
Similar note about casing, I'd expect camelCase here for JavaScript

```suggestion
  errorMessages: MemorableDateErrorMessageLookup;
```

Reply:
Updated `error_messages` and `ranged_errors` to lowerCamelCase.